app/src/main/java/com/arctouch/codechallenge/view/home/HomeActivity.kt

```diff
@@ -0,0 +1,187 @@
+        super.onCreate(savedInstanceState)
+
+        mViewModel = createViewModel()
+        configureDataBinding()
+        registerReceiver(mConnectivityReceiver, IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION))
+    }
+
+    private fun configureDataBinding() {
+        mBinding = DataBindingUtil.setContentView(this, R.layout.home_activity)
+        mBinding.lifecycleOwner = this
+
+        val adapter = HomeAdapter(emptyList<Movie>().toMutableList())
+        adapter.listener = this
+        mBinding.recyclerView.adapter = adapter
+
+        this.lifecycle.addObserver(mViewModel)
+        mBinding.viewModel = mViewModel
+
+        mBinding.recyclerView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrolled(recyclerView: RecyclerView?, dx: Int, dy: Int) {
+                super.onScrolled(recyclerView, dx, dy)
+                if (!recyclerView!!.canScrollVertically(1)
+                        && !mIsLoading) {
+                    mBinding.recyclerView.post { adapter.showLoading() }
+                    val dispose = mViewModel.requestNextPage().subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
+                            .subscribe {
+                                adapter.hideLoading()
+                            }
+
+                    mCompositeDisposable.add(dispose)
+
+
+                }
+            }
+        })
+    }
+
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        if (intent != null) {
+            handleIntent(intent)
+        }
+    }
+
+    private fun handleIntent(intent: Intent) {
+
+        if (Intent.ACTION_SEARCH == intent.action) {
+            val query = intent.getStringExtra(SearchManager.QUERY)
+//            Toast.makeText(this, query, Toast.LENGTH_SHORT).show()
+            mViewModel.searchMovie(query)
+            //use the query to search your data somehow
+
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.options_menu, menu)
+        val searchManager = getSystemService(Context.SEARCH_SERVICE) as SearchManager
+        val menuItem = menu.findItem(R.id.search)
+        val searchView = menuItem.actionView as SearchView
+        searchView.apply {
+            setSearchableInfo(searchManager.getSearchableInfo(componentName))
+        }
+        menuItem.setOnActionExpandListener(object : MenuItem.OnActionExpandListener {
+            override fun onMenuItemActionExpand(item: MenuItem?): Boolean {
+
+                return true
+            }
+
+            override fun onMenuItemActionCollapse(item: MenuItem?): Boolean {
+                mViewModel.resetSearchVariables()
+                return true
+            }
+
+        })
+
+        return true
+    }
+
+    private fun createViewModel(): MoviesViewModel {
+
+        val movieService = MovieService(TmdbApi.create())
```

Comment:
Here you could have made all variables as `lazy` as well. Doing this, you can better mock them out and make the class more testable. An alternative code could be written like this:

    val movieService by lazy { MovieService(TmdbApi.create()) }
    val movieRepository by lazy{ MovieRepository(movieService) }
    val viewModel  by lazy { MoviesViewModel(application, movieRepository) } 

Another alternative should be to let viewModel as `var lateinit`, just to make clear that it will be initialized later on below.